pkg/telemetry/tracing.go

```diff
@@ -100,7 +101,12 @@ func MustNewTracerProvider(opts ...TracerOption) *sdktrace.TracerProvider {
 	return tp
 }
 
+// TraceError marks the span as having an error, except if the error is context.Canceled,
+// in which case it does nothing.
 func TraceError(span trace.Span, err error) {
+	if errors.Is(err, context.Canceled) {
+		return
+	}
 	span.RecordError(err)
```

Comment:
Wondering if this is the right choice? My intuition suggests that it would be good to see the spans which have been cancelled as that will help us identify ongoing bodies of work that were cancelled due to upstream resolution having occurred already.

Reply:
The reason I added this is because I don't want to pollute the trace with errors when the behavior you mention is a feature of how the code works. I want to mark the span as errored when it is an actual error